app/main.py

```diff
@@ -1,13 +1,45 @@
 def format_linter_error(error: dict) -> dict:
-    # write your code here
-    pass
+    return {
+        "line": error["line_number"],
+        "column": error["column_number"],
+        "message": error["text"],
+        "name": error["code"],
+        "source": "flake8"
+    }
 
 
 def format_single_linter_file(file_path: str, errors: list) -> dict:
-    # write your code here
-    pass
+    return {
+        "errors": [
+            {
+                "line": error["line_number"],
+                "column": error["column_number"],
+                "message": error["text"],
+                "name": error["code"],
+                "source": "flake8"
+            }
+            for error in errors
+        ] if errors else [],
+        "path": file_path,
+        "status": "failed" if errors else "passed",
+    }
 
 
 def format_linter_report(linter_report: dict) -> list:
-    # write your code here
-    pass
+    return [
+        {
+            "errors": [
+            {
+                "line": mistake["line_number"],
+                "column": mistake["column_number"],
+                "message": mistake["text"],
+                "name": mistake["code"],
+                "source": "flake8"
```

Comment:
This block of code is also duplicated in the `format_linter_error` function. Consider using the `format_linter_error` function to avoid repetition and adhere to the 'Don't Repeat Yourself' principle.